### PR TITLE
Heap sort using STL functions

### PR DESCRIPTION
**Pull request by deepaks2112**
Deepak Singh
11740290
deepaks@iitbhilai.ac.in

### CODE REVIEW

**Comment by shashwatj07** [comment]
@yatendra1999 another student has already submitted a PR for this function. How to tackle these kind of scenarios?
Also, I don't recommend using STL here. What say?

**Comment by yatendra1999** [comment]
Whichever PR is ready to be merged first takes priority. Usage of STL functions is not recommended. Some STL usage is still allowed for data structures and trivial algoritms. In this case, it's not trivial and not allowed.

File: Sorting/heap.cpp [hunk 1]
```diff
@@ -8,5 +9,17 @@
...
  */
 void Sort::HeapSort(){
-    std::cout<<"[TODO @ Sorting/heap.cpp] HeapSort() needs to be implemented.\n";
-}
\ No newline at end of file
+    std::cout << "Sorting the array using heap sort." << std::endl;
+
+    sorted = unsorted;
+	size_t curr_size = sorted.size();
+	make_heap(sorted.begin(), sorted.begin() + curr_size);
```
Comment (yatendra1999) [inline Sorting/heap.cpp:16]: Do not depend on STL to skip major parts of the algorithm.

File: Sorting/heap.cpp [hunk 2]
```diff
@@ -8,5 +9,17 @@
...
+    
+    while(curr_size > 0){
+
+    	pop_heap(sorted.begin(), sorted.begin() + curr_size);
```
Comment (yatendra1999) [inline Sorting/heap.cpp:20]: Same.

**Review by yatendra1999 — CHANGES_REQUESTED** [review]
more work is needed.

**Pull request closed without merging**